src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -340,8 +343,66 @@ private async Task<ActionResult> AuthenticateMaskinportenToken(string originalTo
                     }
                 }
 
+                string authenticatemethod = "maskinporten";
+
+                if (!string.IsNullOrEmpty(Request.Headers["X-Altinn-EnterpriseUser-Authentication"]))
+                {
+                    string enterpriseUserHeader = Request.Headers["X-Altinn-EnterpriseUser-Authentication"];
+                    string decodedString;
+
+                    try
+                    {
+                        decodedString = DecodeEnterpriseUserHeader(enterpriseUserHeader);
+                    }
+                    catch (Exception)
+                    {
+                        return StatusCode(400);
+                    }
+
+                    string[] decodedStringArray = decodedString.Split(":");
+                    string usernameFromRequest = decodedStringArray[0];
+                    string password = decodedStringArray[1];
```

Comment:
This will throw NullReferenceException if there's no ':' in the input string. As mentioned in the previous comment, the helper should do all the parsing of this header - not just the base64-decoding. 
```
private (string, string) DecodeEnterpriseUserHeader(string enterpriseUserHeader) { ... }

...

string userName, password;
try {
   (username, password) = DecodeEnterpriseUserHeader(Request.Headers["X-Altinn-EnterpriseUser-Authentication"])
}
catch (Exception e) {
    return StatusCode(400);
}
...
```

Reply:
You are right, I missed that. I have now moved the rest of the string-handling related to credentials-decoding into the helpermethod.